src/transactions/TransactionFrame.cpp

```diff
@@ -813,6 +813,8 @@ TransactionFrame::refundSorobanFee(AbstractLedgerTxn& ltxOuter)
         return;
     }
 
+    getResult().feeCharged -= feeRefund;
```

Comment:
No chance of feeCharged going negative here, right?

Reply:
There should be no way for this to happen because we always initialize feeCharged to the charged fee. If it were to go negative due to a bug (feeCharged is a int64), it wouldn't be catastrophic from a core perspective because we don't use the value, but would definitely need to be fixed.